### ISSUE TITLE
Twisted Metal: Head-On no sfx

### ISSUE THREAD

**Issue by Godavatar** [hrydgard/ppsspp#10933]
I know this was fixed in 1.5.4, but for some reason there no sounds in game except music and cars special weapons. I tried to change every setting in case this helps, but issue still exist. I am using the last build (9.15).

**Comment by unknownbrackets** [comment]
If you go back to 1.5.4, does it work?  Or is it not working for you there?

Assuming it does work, can you find the git build where it stopped working?  The trick is to go by halves: we're 918 changes away from 1.5.4 currently.  So if you try v1.5.4-484-g22e28e218, and it works there, you know it was broken AFTER 484, so go halfway up from there.  Otherwise, you know it was already broken at 484 and you should try halfway lower.

Doing it this way would take 8-10 tries to narrow it down to which change broke it.  Then we can figure out what was wrong with that change.

-[Unknown]

**Comment by Godavatar** [comment]
So, i tried game on 1.5.4 and, oddly, sfx didnt work. Again tried to change some settings and it didnt help. So i guess later versions dont change much and have same issue in them.

**Comment by unknownbrackets** [comment]
We did make a change in recent git that affected some ways games mix music.  Has anything changed here?

-[Unknown]

**Comment by Ghaleon777** [comment]
> We did make a change in recent git that affected some ways games mix music. Has anything changed here?
> 
> -[Unknown]

No, i tested right now with the newest build and the issue its still there, no fx sound.

**Comment by TAbdiukov** [comment]
Can confirm that as per v1.8.0, the issue still persists

If that's of any help, it seems like if you get the game to play as few sounds as possible, the effects appears (such as engine running), but once there are too many effects, most of the effects cease to play

**Comment by Ghaleon777** [comment]
This problem was always on PPSSPP, i tested older builds (1.5, 1.6, ect) and the issue is always there, i could get it work sometimes, but 70% of the times there is no fx sound. Hope this can be fixed, i love this game.

**Comment by Panderner** [comment]
Same Happens on v1.9.3-906-ge8b737226 but there's no error messages for sound effects

**Comment by Panderner** [comment]
i've turned off the music and set the effect volume max still no SFX

**Comment by Panderner** [comment]
in Latest build, story mode and challenge mode can hear Sound effects but endurance mode doesn't hear sound effects

**Comment by unknownbrackets** [comment]
Is that a change from a previous build?  If so, it might help to know what fixed SFX in some modes, in case it's related.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Godavatar try this game again using the latest ppsspp build from buildbot.

**Comment by hrydgard** [comment]
Don't think this one is related to the MP3 stuff unfortunately.

**Comment by TAbdiukov** [comment]
> 


@Gamemulatorer 
> Godavatar try this game again using the latest ppsspp build from buildbot.

Tried the game on
* The last release build (1.12.3), 
* The last snapshot build (v1.12.3-867-gf58d4dfcf)

The issue still persists

**Comment by albrechtjess** [comment]
So I recently got a large PSP memory card to install games to and this problem is currently happening on my PSP too so not sure if it's an issue with loading the files from flash memory and not a UMD or what.

**Comment by hrydgard** [comment]
Wow that's interesting. Maybe there's some issue with how the game handles free space reporting, not expecting modern sizes, but that affecting sound effects is sure very weird..

**Comment by albrechtjess** [comment]
I know thug2 remix does something similar where it thinks there's not
enough space and has a pop up about it even though there's 24gb free. I
know at the time there was only 2gb cards so it may be a limitation that
was never patched since they were launch titles.

**Comment by unknownbrackets** [comment]
Could someone check if putting the game under `[ReportSmallMemStick]` or `[MemstickFixedFree]` in compat.ini changes anything?

It might also be worth investigating where it checks free space, which would typically be a GETSIZES or SIZE savedata dialog call or code 0x02425818 in sceIoDevctl.

-[Unknown]

**Comment by TAbdiukov** [comment]
> Could someone check if putting the game under [ReportSmallMemStick] or [MemstickFixedFree] in compat.ini changes anything?

@unknownbrackets Nope, no changes whatsoever

> It might also be worth investigating where it checks free space, which would typically be a GETSIZES or SIZE savedata dialog call or code 0x02425818 in sceIoDevctl.

Havent tried this

**Comment by chrisstonehill** [comment]
Just found the same problem on a Retroid Pocket 2+ with v1.12.3 of PPSSPP. Wondering if the just released 1.13.1 will help. Same problem in another game, Unbound Saga...a brief moment of sfx and music when the game begins...then just music from then on. Anyone found a solution?

**Comment by benderscruffy** [comment]
[v1.13.2-2369-gb06ea9784](https://github.com/hrydgard/ppsspp/commit/b06ea9784580302ba3456712a8d2aec770687b1c)
both music and sound fx play on my PC
does anyone still have problems with this game

**Comment by kvg198x** [comment]
> [v1.13.2-2369-gb06ea9784](https://github.com/hrydgard/ppsspp/commit/b06ea9784580302ba3456712a8d2aec770687b1c) both music and sound fx play on my PC does anyone still have problems with this game ??

On v1.14 android; the problem still persists! Also, the 60fps cheat disables the player camera meaning it doesn't follow the car and remains in a fixed position.

**Comment by TAbdiukov** [comment]
JFYI, PPSSPP v1.17 - issue is still present

**Comment by JoaoVicthor** [comment]
Just saw the following fix in [a reddit thread](https://www.reddit.com/r/TwistedMetal/comments/15j9byv/twisted_metal_head_on_psp_sounds_fix/).
It fixes the issue for me.
```
Game Settings ---> System ---> I/O timing method and select "Simulate UMD delays".

Restart the game.
```

**Comment by hrydgard** [comment]
@JoaoVicthor Great find! I'll make sure we always enabled it for this game.

**Comment by TAbdiukov** [comment]
I'm confirming that the fix works -- awesome job!

**Comment by hrydgard** [comment]
Turns out, not very surprisingly, that this only partially fixes the problem. "open" on Discord reports that on the second level, some sounds disappear again.

Given the symptoms, it's very likely that the game itself has a synchronization bug, making it dependent on the exact time some operation takes in relation to some other operation happening at the same time. Unfortunately the timings of various things on the PSP (including UMD accesses, even with delays) are not very accurately emulated by PPSSPP, leading to this type of problem, and the fix not really fixing things completely...

**Comment by chrisstonehill** [comment]
I found a difference if I had Skip buffer effects on or not. With it off,
when I did a save state at the start of the 2nd level...and thus a
thumbnail showing me where I was...when starting from the Save State
another time, sound effects were gone again. BUT, if I turned Skip buffer
effects on and saved at the start of the 2nd level, a completely new play
session...just a black thumbnail in the save state menu...but sound effects
maintained 👍👍👍

**Comment by hrydgard** [comment]
Hm, I doubt skip buffer effects actually affected it, more likely to be random...

**Comment by chrisstonehill** [comment]
OK...thanks for clarification. Whether it is on or not has interesting
consequences to Ultimate Ghosts and Goblins. Works with both, but some
situations result in lots of slowdown...when it is off, I think.

...and of course some games you have to have it...off?...in order to
display anything, right?

**Comment by chrisstonehill** [comment]
...such as Street Fighter Alpha 3 MAX

**Comment by Troy168** [comment]
I might as well throw my two cents in since I was also experiencing the same issue as well. But it seems like I found a work around at least for now. When you're playing through the game and select the next map, the game will have a bug where the SFX would shut off again. But if you chose to leave after saving your game and then loading it back in, the game run as normal.

It's a pain, sure. But it's the only workaround I can find until this SFX bug is fully fixed.

**Comment by Troy168** [comment]
Went further down in the game and now there's the issue when reaching the last level in story mode, the sound effects just wouldn't play. Even when the setting is set to UMD delay emulation, the sound effects would not play. I've also went into the other game modes and they still have this issue with no sound effects despite the settings being set correctly.

**Comment by ninjz** [comment]
Same I've tried everything. I'm on 1.18.1 through Knulli on a rg34xxsp

Update: I actually got it working but using a different setting "Simulate UMD slow reading speed" instead of "Simulate UMD delays". Not sure why but it works!

**Comment by Troy168** [comment]
I have tested what ninjz suggested and I can confirm it does work on all game modes.
